src/blog/2025/01/integrating-siemens-s7-plcs-with-node-red-guide.md

```diff
@@ -0,0 +1,237 @@
+
+<!--more-->
+
+Siemens S7 PLCs are typically programmed using TIA Portal, Siemens' integrated development environment, and communication with external systems usually relies on the S7 protocol (ISO over TCP/IP). In this article, we’ll walk you through how to use Node-RED to read from and write to Siemens S7 PLCs via the S7 protocol, unlocking new possibilities for remote control and system integration in your industrial automation setup.
+
+## Prequsite
+
+Before integrating your Siemens S7 PLC with Node-RED, make sure you have the following :
+
+1. Before downloading the ladder program and all configurations and settings to your PLC, make sure you have the following settings:
+
+- Allow PUT/GET Communication from remote partners.
+
+![PUT/GET Communication from remote partners is Allowed](./images/allow-put-get-communication.png){data-zoomable}
+_PUT/GET Communication from remote partners is Allowed_
+
+- Provide full access to the PLC (no protection), allowing unrestricted access to data exchange.
+
+![Providing complete access to the PLC](./images/providing-full-access-to-plc.png){data-zoomable}
+_Providing complete access to the PLC_
+
+2. Ensure that the appropriate ladder program (or any other logic) is written according to your requirements and successfully downloaded to the PLC. However, before downloading, make sure the 'Optimized Block Access' option is disabled for the data block that your ladder program using.
+
+![Untick 'Optimized Block Access'.](./images/optimized-block-access.png){data-zoomable}
+_Untick 'Optimized Block Access.'_
+
+3. Install Node-RED on the device that will communicate with the S7 PLC. You cannot install Node-RED directly on the S7 PLC, as PLCs are typically controllers, not computers. For example, you can use a device like the Revolutionary Pi to connect and transfer data across systems. Use the [FlowFuse Device Agent](/product/device-agent/) to install Node-RED on your device. 
+
+- Why FlowFuse Device Agent? It allows you to manage Node-RED remotely, enabling control, monitoring, and flow creation without the need for on-site visits. FlowFuse also offers a suite of enterprise-grade features such as collaboration, device management, and DevOps pipelines, which are essential in industrial environments. These features help streamline operations and ensure scalability in complex automation systems. For setup, refer to the [installation guide](/docs/device-agent/install/).
+
+4. Verify that the device running Node-RED is in the same network as the PLC and can successfully ping the PLC. Also, a firewall should not block the S7 port (typically port 102).
+
+```bash
+ping <PLC_IP>
+```
+
+```bash
+netstat -an | findstr 102
+```
```

Comment:
should we make a nod to the fact this that PING can be done inside node-red using `node-red-node-ping`?

also, the exec node can do these commands

also, I `findstr` is windows only. on linux it would be `grep` 

rather than provide all the different ways for the different OS's, perhaps simplify to ping only and mention firewalls as a side note?